### PR TITLE
Added support for "disable: 'if-port-is-already-open'"

### PR DESCRIPTION
**Pull request by shyiko**
Hello Alex,

as you already know PR introduces 'if-port-is-already-open' as a possible value for httpServer.disable/selenium.disable options. Please note that it's missing sauceConnect.disable (Sauce Connect listens on port 4445 be default). I would have added it myself by I decided to opt for another solution in order to be able to run tests in parallel (strkio/strkio.github.io@449e8589a11f60496a4ef3a3029d5bed8d718867).

Anyway, thanks a lot for the project. It's been immensely helpful.

### CODE REVIEW

**Comment by alexbrombal** [comment]
I just have two questions about it:

- I like the addition of the async library, though I wonder if we should use `async.series` instead of `async.parallel`. If the setup for each component (http server, sauce labs, sauce connect, etc) happens in parallel, there may be unexpected race conditions; or more obviously, the console output will be hard to follow because they happen out of order. 
- What should happen if the port is in use if you *don't* set `disabled: "if-port-is-already-open"`? To keep it simple, I wonder if what you implemented should just be the standard behavior (i.e. skip whatever component needs that port). Right now the entire process just errors and quits, but as long as there is information in the console output, I don't see that being a common use case.

What do you think?

**Comment by shyiko** [comment]
> If the setup for each component (http server, sauce labs, sauce connect, etc) happens in parallel, there may be unexpected race conditions

There should be no race conditions because all three of them (http server, selenium and sauce connect) are completely independent.

> the console output will be hard to follow because they happen out of order

True. But, personally, I would choose "less time to wait" over "log consistency". 

> I wonder if what you implemented should just be the standard behavior

I didn't want to introduce backward-incompatible changes. That's all. I have no way of knowing whether somebody is using 'selenium-sauce' and expecting it to fail, let's say, if selenium port if open. But then again, I would have made it the default behavior if I was the one to decide.

**Comment by alexbrombal** [comment]
Thanks for the feedback. I'll probably go ahead and make it the default behavior. It's still in beta and I don't think enough people are using it to worry about making backward incompatible changes yet.

**Comment by shyiko** [comment]
Thank you, Alex.
